lib/audumbla/enrichments/coarse_geocode.rb

```diff
@@ -0,0 +1,203 @@
+    # @param [#lat#long] place  a place to verify a match against
+    #
+    # @result [Boolean] true if the interpretation is accepted
+    def match?(interpretation, place)
+      return true if place.lat.empty? || place.long.empty?
+
+      point = Geokit::LatLng.new(place.lat.first, place.long.first)
+      if interpretation.geometry.bounds.nil?
+        # measure distance between point centers
+        distance = twofishes_point_to_geokit(interpretation.geometry.center)
+                   .distance_to(point, unit: :kms)
+        return distance < DEFAULT_DISTANCE_THRESHOLD_KMS
+      end
+        
+      twofishes_bounds_to_geokit(interpretation.geometry.bounds)
+        .contains?(point)
+    end
+
+    private
+
+    ##
+    # Populates a DPLA::MAP::Place with data from a given feature. This 
+    # overwrites existing data with the exception of the identity (URI or node 
+    # id) and the `providedLabel`. `exactMatch`, `closeMatch`, `label` 
+    # (skos:prefLabel)and all other geographic data is replaced.
+    # 
+    # @param [DPLA::MAP::Place] place  a place to enrich
+    # @param [GeocodeFeature] feature  a twofishes feature whose data should be
+    #   added to place.
+    #
+    # @return [DPLA::MAP::Place] the original place enriched
+    def enrich_place(place, feature)
+      place.label = feature.display_name
+      place.exactMatch = feature_to_geoname_uris(feature)
+      place.closeMatch = feature_to_close_matches(feature, 
+                                                  /^http\:\/\/id\.loc\.gov\/.*/)
+      place.countryCode = feature.cc
+      place.lat = feature.geometry.center.lat
+      place.long = feature.geometry.center.lng
+
+      place
+    end
+
+    ##
+    # Extracts geonameids for the given feature and converts them into URIs
+    #
+    # @param [GeocodeFeature] feature the feature to identify
+    #
+    # @return [Array<RDF::URI>] a list of geoname URIs. Generally, this will only 
+    #   contain one exactly matching geonameid in URI form.
+    def feature_to_geoname_uris(feature)
+      geoname_ids = feature.ids.select { |id| id.source == :geonameid.to_s }
+      geoname_ids.map { |id| RDF::URI('http://sws.geonames.org') / id.id + '/' }
+    end
+
+    ##
+    # Extracts URIs for closely matching terms in other authority or knowledege
+    # organization systems
+    #
+    # @param [GeocodeFeature] feature the feature to identify
+    # @param [Regexp] patterns a splat argument containing any number of 
+    #   patterns matching
+    #
+    # @return [Array<RDF::URI>] a list of matching ids
+    def feature_to_close_matches(feature, *patterns)
+      union = Regexp.union(patterns)
+      feature.attributes.urls.select { |str| union.match(str) }
+        .map { |id| RDF::URI(id) }
+    end
+    
+    ##
+    # Sends a geocode request. This is used in lieu of `Twofishes#geocode`, 
+    # since that method does not allow passing parameters other than 
```

Comment:
We could. It seems pretty deliberate, though. There are a bunch of other configuration options that Twofishes exposes that aren't there.

I half expect we'd hear back PR's accepted---and maybe we should follow through if so. Let's talk about this at check-in tomorrow?